app/models/user.rb

```diff
@@ -1,6 +1,22 @@
 class User < ActiveRecord::Base
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable, :confirmable
+  devise :database_authenticatable, :registerable, :recoverable, :rememberable,
+         :trackable, :validatable, :confirmable, :omniauthable, omniauth_providers: [:facebook]
+
+  FACEBOOK_GENDERS = { 'male' => 1, 'female' => 2 }
+
+  def self.from_omniauth(auth)
+    where( provider: auth.provider, uid: auth.uid ).first_or_create do |user|
+      user.email = auth.info.email
+      user.password = Devise.friendly_token[0,20]
+      user.name = auth.info.name
+      if FACEBOOK_GENDERS.key?( auth.extra.raw_info.gender )
+        user.gender = FACEBOOK_GENDERS[auth.extra.raw_info.gender]
```

Comment:
Declare `gender` as `enum` and then check if the facebook gender matches any of ours.

Reply:
Also, write it in one line, something like `user.gender = genders.key(auth.extra.raw_info.gender) ? auth.extra.raw_info.gender : 'others'`